### DISCUSSION TITLE
dependency OS specific

### DISCUSSION THREAD

**Discussion by ebrahimraeyat** [xcfem/xc#108] [Q&A] [answered]
Hello @lcpt. Thank you for great software.

I want to install xc on windows natively, not by WSL technology.
what is the Linux specific libraries?

**Comment by lcpt** [comment]
Hi, @ebrahimraeyat!

I'm not sure. Some years ago, I tried to compile it on Windows through [cygwin](https://www.cygwin.com/), but some libraries were missing. I've never tried again.

**Comment by ebrahimraeyat** [comment]
Thanks, but I want to know if really possible to prepare a ready to use .exe file for users. I track your work and as you know, It is a very good software for engineers, But I think because not possible to install it on windows, it is not use widely by engineers.
Unfortunately I don't know about some dependency packages, but seems most of those have windows version. for others (I don't know which), maybe possible to use equivalent. I want to know which dependency packages is specific to Linux and don't have equivalent windows package?

**Comment by lcpt** [comment]
Sorry @ebrahimraeyat, I don't know. I barely use Windows once in a month, even less, so I know very little about programming libraries available for this platform.

On the other hand, I think true Windows users are unwilling to use software that forces them to write Python scripts (or any other form of programming). They prefer black boxes with plenty of dialog boxes :)

**Comment by ebrahimraeyat** [comment]
Thanks. Yes, I agree with you. But I think if it is available for windows, other developers like me(!) can write Gui for it. I am very interested to write an open source Gui for xcfem, But I think it is require a massive work on it and thus  time consuming. If it be available for windows, during developement, I can sell some part to be able to continue the remaining project.

BTW, I searched for dependencies. I don't know if I am right.

# XC Dependencies.

## API. User interface
 - [x] Python
 - [x] Visualization ToolKit (VTK)
 - [x] LaTeX
 - [x] Matplotlib
 - [x] ImageMagick
 - [x] Mayavi
 - [ ] C++ Template Image Processing Toolkit -> optional?
 - [x] Open Graphics Library
 - [ ] X Window System] -> is it needed in windows?

## Computational geometry. Mesh generation

 - [x] Computational geometry algorithms library (CGAL)
 - [x] Gmsh

## Number crunching

 - [x] OpenSees
 - [x] Boost c++ libraries
 - [x] NumPy
 - [x] SciPy
 - [ ] libmpfr -> what is usage?
 - [x] libgmp
 - [ ] Arpack : because it is Fortran subroutines, it might run on windows
 - [ ] Arpack++ : ?
 - [ ] Lapack Library of linear algebra routines: I can't open the link
 - [ ] SuperLU : This is Ubuntu package?
 - [ ] Suitesparse
 - [ ] PETSC : I don't know, what is it?
 - [ ] Skypack : I don't know, what is it?
 - [ ] ITPACK : I  can't open the site
 - [ ] MUMPS: I don't know
 
## Data storage

 - [ ] Berkeley DB
 - [x] SQLite

## Libraries building. Version control.

- [x] Git
- [ ] GNU C++ compiler : may be The Cygwin, MinGW
- [ ] GNU FORTRAN compiler : maybe the equivalent
- [x] CMake
thanks.

**Comment by lcpt** [comment]
Yes, those are the dependencies. I've fixed the ITPACK and some other broken links in [the dependencies list](https://github.com/xcfem/xc/blob/master/install/dependencies_list.md), thanks. This link and its neighbors correspond to linear algebra libraries.

**Reply by ebrahimraeyat** [reply]
Thanks. The dependencies that checked are available in windows. I want to know if we can replace (in theory and really) others that not available for windows? for example using numpy for linear algebra? or maybe the fortran is faster? Thanks.

**Reply by lcpt** [reply]
In theory, you can replace whatever you want (any data in a computer is a sequence of binary digits) in practice the changes you propose, in my opinion, make not sense. The solution algorithms XC inherits from OpenSees are a very well designed set of routines to deal with a variety of problems. Replacing the linear algebra libraries with callings to numpy libraries is not obvious at all, and it will be certainly slower.

**Reply by ebrahimraeyat** [reply]
Thanks, with theory I mean the time consuming and widely use package in xc, not really in theory. Thanks.

**Comment by smohaorg** [comment]
I would also like to give a hand but I don't own a Windows machine. I can try to cross-compile it on Debian and then check if it works on Windows or not. 

Thank you @ebrahimraeyat for starting this thread. If you prepare a comprehensive list of all needed packages, I can start working on it.

**Reply by ebrahimraeyat** [reply]
Hello @smohaorg. I am so happy to see you in this thread (I and @smohaorg are from same country!)

the complete list of dependencies are in [above](https://github.com/xcfem/xc/discussions/108#discussioncomment-2499235) post ([original link](https://github.com/xcfem/xc/blob/master/install/dependencies_list.md)), I checked the libraries that presents for windows.

Thanks.

**Comment by smohaorg** [comment]
But @ebrahimraeyat if you want to sell your compiled package, the first thing you have to do is removing OpenSees. Maybe we can write an extra interface and let the end users download OpenSees.

If we don't do it we will have license compatibilty problems. Because OpenSees is a proprierary sofware with no open-source license.

**Comment by ebrahimraeyat** [comment]
Yes @smohaorg. And as XCFEM uses OpenSees source code directly, I think we can't tell the user to only download the executable opensees, because it did not work.

**Reply by lcpt** [reply]
The idea was to develop a powerful analysis tool that was useful in the daily work of a structural engineering practice. The possibility of making a closed binary package from it was clearly and intentionally disregarded.

This is all about freedom, not about money.

**Reply by smohaorg** [reply]
This is exactly what I could guess but I meant that OpenSees license is not GPL-compatible. As far as I know, if Ebrahim wants to compile everything into one executable file, then he cannot release it under GPL. He has to strip OpenSees first.

**Reply by lcpt** [reply]
That is not impossible, but almost.

**Reply by ebrahimraeyat** [reply]
I don't want to close the source, I want an open source Gui. But we can sell it to be able to continue developing or maybe get donates from users.

**Comment by lcpt** [comment, marked as answer]
I understand. Anyway, I think the easiest way to get the software running on Windows is to solve the problem with the X window server (on the Windows side).

**Reply by ebrahimraeyat** [reply]
I tried but not succeed. maybe I must get windows 11 a try.

**Reply by ebrahimraeyat** [reply]
> I tried but not succeed. maybe I must get windows 11 a try.

solved: https://github.com/xcfem/xc/issues/105#issuecomment-1470488714
